api/api/test/test_validator.py

```diff
@@ -0,0 +1,172 @@
+])
+def test_validation_check_exp_ko(exp, regex_name):
+    assert validator.check_exp(exp, regex_name) is False
+
+
+@pytest.mark.parametrize('relative_path', [
+    ('etc/rules/new_rule.xml'),
+    ('etc/decoders/new_decoder.xml'),
+    ('etc/lists/new_list'),
+    ('etc/ossec.conf')
+])
+def test_validation_paths_ok(relative_path):
+    assert validator.check_path(relative_path) is True
+
+
+@pytest.mark.parametrize('relative_path', [
+    ('etc/rules/new_rule'),
+    ('../../bin'),
+    ('etc/decoders/decoder'),
+    ('etc/internal_options')
+])
+def test_validation_paths_ko(relative_path):
+    assert validator.check_path(relative_path) is False
+
+
+@pytest.mark.parametrize('parameters, filters', [
+    ({'path': 'etc/rules/rule.xml', 'offset': '32', 'format': 'xml'},
+     {'path': 'paths', 'offset': 'numbers', 'limit': 'numbers', 'format': 'type_format'}),
+    ({'offset': '2', 'limit': '3', 'sort': '-agent_id'},
+     {'offset': 'numbers', 'limit': 'numbers', 'sort': 'sort_param'}),
+    ({'ip': '192.168.122.15', 'os.name': 'CentOS Linux', 'os.version': '7.1'},
+     {'ip': 'ips', 'os.name': 'alphanumeric_param', 'os.version': 'alphanumeric_param'}),
+    ({'use_http': 'true', 'force': '1', 'agent_id': '004'},
+     {'use_http': 'boolean', 'force': 'numbers', 'agent_id': 'numbers'})
+])
+def test_validation_check_parms_ok(parameters, filters):
+    assert validator.check_params(parameters, filters) is True
+
+
+@pytest.mark.parametrize('parameters, filters', [
+    ({'path': 'etc/internal_options', 'offset': '32', 'format': 'xml'},
+     {'path': 'relative_paths', 'offset': 'numbers', 'limit': 'numbers', 'format': 'type_format'}),
+    ({'offset': '2', 'limit': '3', 'sort': '-agent_id', 'path': 'etc/rules/local_rules.xml'},
+     {'offset': 'numbers', 'limit': 'numbers', 'sort': 'sort_param'}),
+    ({'ip': '192.168.122.345', 'os.name': 'CentOS Linux', 'os.version': '7.1'},
+     {'ip': 'ips', 'os.name': 'alphanumeric_param', 'os.version': 'alphanumeric_param'}),
+    ({'use_http': 'true', 'force': '1', 'agent_id': '004'},
+     {'use_http': 'boolean', 'force': 'numbers'})
+])
+def test_validation_check_parms_ko(parameters, filters):
+    assert validator.check_params(parameters, filters) is False
+
+
+@pytest.mark.parametrize('cdb_list', [
+    ('audit-wazuh-w:write \n audit-wazuh-r:read \n audit-wazuh-a:attribute \n'
+     'audit-wazuh-x:execute \n audit-wazuh-c:command')
+])
+def test_validation_cdb_list_ok(cdb_list):
+    assert validator.check_cdb_list(cdb_list) is True
+
+
+@pytest.mark.parametrize('cdb_list', [
+    (':write \n audit-wazuh-r:read \n audit-wazuh-a:attribute'),
+    ('audit-wazuh:write \n $variable:read \n audit-wazuh-a:attribute')
+])
+def test_validation_cdb_list_ko(cdb_list):
+    assert validator.check_cdb_list(cdb_list) is False
+
+
+@pytest.mark.parametrize('xml_string', [
+    ('<!--  Wazuh - Manager - Default configuration for centos 7  More info at: '
```

Comment:
Please, take this embedded xml file to a test/data/test.xml path to make this test cleaner.

Reply:
Done!